Sources/MapboxDirections/TollPrice.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+import Turf
+
+/**
+ :nodoc:
+ Information about toll payment method.
+ */
+public struct PaymentMethod: Hashable, Equatable {
+    /**
+     Method identifier.
+     */
+    public let identifier: String
+    
+    /**
+     Payment is done by electronic toll collection.
+     */
+    public static let electronicTollCollection = PaymentMethod(identifier: "etc")
+    /**
+     Payment is done by cash.
+     */
+    public static let cash = PaymentMethod(identifier: "cash")
+}
+
+/**
+ :nodoc:
+ Categories by which toll fees are divided.
+ */
+public struct Category: Hashable, Equatable {
+    /**
+     Category name.
+     */
+    public let name: String
+    
+    /**
+     A small sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let small = Category(name: "small")
+    /**
+     A standard sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let standard = Category(name: "standard")
+    /**
+     A middle sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let middle = Category(name: "middle")
+    /**
+     A large sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let large = Category(name: "large")
+    /**
+     A jumbo sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let jumbo = Category(name: "jumbo")
+}
+
+public typealias CategoriesTolls = [Category: Decimal]
+public typealias PaymentMethods = [PaymentMethod: CategoriesTolls]
+
+/**
+ :nodoc:
+ Toll cost information for the `Route`.
+ */
+public struct TollPrice: Equatable,  Codable, ForeignMemberContainer {
+    public var foreignMembers: Turf.JSONObject = [:]
+    
+    private enum CodingKeys: String, CodingKey {
+        case currency
+        case paymentMethods = "payment_methods"
+    }
+
+    /**
+     Related currency code string.
+     
+     Uses ISO 4217 format. Refers to values in `CategoriesTolls`. A toll cost of `0` is valid and simply means that no toll costs are incurred for this route.
```

Comment:
As a developer, I’d expect to be able to plug this `currencyCode` directly into `NumberFormatter.currencyCode`, but this special case would be surprising and easy to miss. If a road is free, either there should be no toll price (`nil` entry in the array), or if we need to distinguish this from an unknown value, then the currency code should be `nil` and any payment amount should be 0. We can always convert the `nil` to a `0` string when converting back to JSON.

(This would be a little less awkward if the tolls were structured as a dictionary containing structs containing currency codes and amounts, rather than a struct containing currency codes and dictionaries of amounts.)

Reply:
Documentation is sparse on that point, but it never said that currency code may be empty. I expect that it will still be local code even if the route is free.